packages/cdktf/lib/backends/local-backend.ts

```diff
@@ -14,6 +15,19 @@ export class LocalBackend extends TerraformBackend {
   protected synthesizeAttributes(): { [name: string]: any } {
     return keysToSnakeCase({ ...this.props });
   }
+
+  public getRemoteStateDataSource(
+    scope: Construct,
+    name: string,
+    fromStack: string
+  ): TerraformRemoteState {
+    return new DataTerraformRemoteStateLocal(scope, name, {
+      workspace: this.props.workspaceDir,
+      path:
+        this.props.path ||
+        path.join(process.cwd(), `terraform.${fromStack}.tfstate`),
```

Comment:
Theoretically a user could overwrite `terraform.${fromStack}.tfstate` by registering a local backend for their stack explicitly instead of relying on the default local backend which the cdktf-cli uses if no backend has been specified.
Maybe something to add as a future task to improve this?

Reply:
This is why we prefer to take the path configured in the origin backend here and only then default to the default value for the path. It's tested here: https://github.com/hashicorp/terraform-cdk/blob/bf4c591747b2ea5fc21a8817e5ea7d8efe28139c/packages/cdktf/test/app.test.ts#L287

Reply:
Oh, I missed that 👍
However, if the user explicitly specifies the LocalBackend without a path, that [path might default to](https://www.terraform.io/language/settings/backends/local#path) `terraform.tfstate` (as that backend could then take precedence over the legacy `-state` option we pass). But yeah, quite an edge case 🙈

Reply:
Ah related to https://github.com/hashicorp/terraform-cdk/issues/1430 right?